src/mon/Elector.cc

```diff
@@ -566,6 +577,15 @@ void Elector::handle_ping(MonOpRequestRef op)
   dout(10) << __func__ << " " << *m << dendl;
 
   int prank = mon->monmap->get_rank(m->get_source_addr());
+  if (prank >= ssize(mon->monmap->ranks)) {
```

Comment:
I don't think this works -- we just pulled the rank out of the monmap, so if the Mon isn't valid won't we end up with -1 and pass this check, but still fail in the deeper assert?

Reply:
You're right, this isn't valid.